packages/cli/src/ActiveExecutions.ts

```diff
@@ -138,16 +143,13 @@ export class ActiveExecutions {
 			promise.resolve(fullRunData);
 		}
 
-		// Remove from the list of active executions
-		delete this.activeExecutions[executionId];
-
-		this.concurrencyControl.release({ mode: execution.executionData.executionMode });
+		this.postExecuteCleanup(executionId);
```

Comment:
I find the naming of this new method `postExecuteCleanup` confusing because we also call `ActiveExecutions.remove` when putting an execution to wait. Plus we're skipping the `onPostWorkflowExecute` hook for cancellations in the understanding that a cancellation did not reach completion, but at the same time we're also we're calling `postExecuteCleanup` for cancellations.

Maybe simply `cleanup`?

Reply:
I'm gonna do some further refactor and try to add more tests in another PR.
Merged this already to get this out in tomorrow's release.